aquadoggo/src/graphql/queries/all_documents.rs

```diff
@@ -59,26 +58,35 @@ pub fn build_all_documents_query(query: Object, schema: &Schema) -> Object {
                         &mut filter,
                     )?;
 
+                    // @TODO: We need a way to determine which fields have been selected in the
+                    // GraphQL query (is it "lookahead")?
```

Comment:
[`ctx.lookahead()`](https://docs.rs/async-graphql/latest/async_graphql/dynamic/struct.ResolverContext.html#method.look_ahead) tells you inspect query tree [`ctx.field().selection_set()`](https://docs.rs/async-graphql/latest/async_graphql/context/struct.SelectionField.html#method.selection_set) gives you all subfields of the current query level.

Reply:
"selection_set" example here: https://docs.rs/async-graphql/latest/async_graphql/dynamic/struct.ResolverContext.html#method.field

Reply:
(i think we need "selection_set" here)

Reply:
Thank you! There's a couple of places where I need to touch the GraphQL API and I think it's easier to hack on it together in a session? 👍🏻

Reply:
I'm getting some errors when working with relations which might be fast to solve for you. I'm not having my computer around right now but happy to show it in a meeting.

Also we need to look at the pagination API again as I thought that `documents` is an array and the pagination info is an object while currently everything is part of the array.

This is what I would expect as an response:

```
{
  totalCount: 12,
  hasNextPage: true,
  documents: [
    {
      fields: ...
    },
    ...
  ],
}
```

And lastly I'd love to add some additional fields for backwards pagination 👍🏻

Reply:
Cool, I could look into this on Tuesday for sure, and some co-hacking would be great then too.